packages/backend/src/api/controllers/status/StatusController.ts

```diff
@@ -1,119 +1,120 @@
+                status: fromTimestamp(latest?.timestamp),
+              }
+            })
+            .sort(this.isSyncedDesc) ?? [],
       })),
     )
+    return renderBalancesPage({ balances })
+  }
+
+  private isSyncedDesc(a: { status: Status }, b: { status: Status }) {
+    return +a.status.isSynced - +b.status.isSynced
   }
 
-  async getReportsStatus() {
+  async getReportsStatus(): Promise<string> {
     const bridgeLatest = await this.reportsRepository.getLatestPerBridge()
-    return this.projects.flatMap(({ bridges, name }) =>
+    const reports = this.projects.flatMap(({ bridges, name }) =>
       bridges.map(({ address }) => ({
-        name,
-        address,
+        projectName: name,
+        holderAddress: address,
         tokens:
-          bridgeLatest.get(EthereumAddress(address))?.map((latest) => ({
-            assetId: latest.asset,
-            balance: latest.balance.toString(),
-            usd: latest.balanceUsd.toString(),
-            eth: latest.balanceEth.toString(),
-            blockNumber: latest.blockNumber.toString(),
-            timestamp: unixTimeToString(latest.timestamp),
-            syncStatus: getSyncStatus(latest.timestamp),
-          })) ?? [],
+          bridgeLatest
+            .get(EthereumAddress(address))
+            ?.map((latest) => ({
+              assetId: latest.asset,
+              balance: asNumber(latest.balance, this.getDecimals(latest.asset)),
+              usd: asNumber(latest.balance, 2),
+              eth: asNumber(latest.balance, 6),
+              status: fromTimestamp(latest.timestamp),
+            }))
+            .sort(this.isSyncedDesc) ?? [],
       })),
     )
+    return renderReportsPage({ reports })
   }
-}
-
-function isSynced(timestamp: UnixTime | undefined) {
-  const now = UnixTime.now().add(-1, 'hours').toStartOf('hour')
-  return !!timestamp && now.equals(timestamp)
-}
 
-const unixTimeToString = (date: UnixTime) => {
-  return date.toDate().toString().slice(4, 21)
-}
-
-const SECONDS_PER_HOUR = 60 * 60
-const getSyncStatus = (
-  date: UnixTime,
-): {
-  isSynced: boolean
-  message: string
-} => {
-  const now = UnixTime.now().add(-1, 'hours').toStartOf('hour')
-
-  const diff = now.toNumber() - date.toNumber()
-
-  if (diff === 0) {
-    return { isSynced: true, message: '✔' }
+  private toTokenDecimals(projects: ProjectInfo[]) {
+    const tokenDecimals = new Map<AssetId, number>()
+    for (const project of projects) {
+      for (const bridge of project.bridges) {
+        for (const token of bridge.tokens) {
+          tokenDecimals.set(token.id, token.decimals)
+        }
+      }
+    }
+    return tokenDecimals
   }
 
-  if (diff > 0 && diff < 24 * SECONDS_PER_HOUR) {
-    return {
-      isSynced: false,
-      message: `out of sync for ${diff / SECONDS_PER_HOUR} hour(s)`,
-    }
+  private getDecimals(assetId: AssetId): number {
+    return this.tokenDecimals.get(assetId) ?? 0
```

Comment:
We sometimes use the pattern that if map.get() returnes undefined, we throw `Programmer error`. Do you think it makes sense here?

Reply:
hmm, interesting idea, I am just not sure whether throwing is needed here, because if there is no decimal then it will just output 123456789 instead of e.g. 123.46

Reply:
In this case throwing is not suitable, because it is an expected case (something might not have synced yet).